src/main/java/it/smartcommunitylab/aac/webauthn/controller/WebAuthnRegistrationController.java

```diff
@@ -0,0 +1,195 @@
+package it.smartcommunitylab.aac.webauthn.controller;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import javax.validation.Valid;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.yubico.webauthn.data.AuthenticatorAttestationResponse;
+import com.yubico.webauthn.data.ClientRegistrationExtensionOutputs;
+import com.yubico.webauthn.data.PublicKeyCredential;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.stereotype.Controller;
+import org.springframework.util.StringUtils;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.server.ResponseStatusException;
+
+import io.swagger.v3.oas.annotations.Hidden;
+import it.smartcommunitylab.aac.common.RegistrationException;
+import it.smartcommunitylab.aac.core.provider.ProviderRepository;
+import it.smartcommunitylab.aac.model.Subject;
+import it.smartcommunitylab.aac.webauthn.WebAuthnIdentityAuthority;
+import it.smartcommunitylab.aac.webauthn.model.WebAuthnAttestationResponse;
+import it.smartcommunitylab.aac.webauthn.model.WebAuthnRegistrationResponse;
+import it.smartcommunitylab.aac.webauthn.model.WebAuthnRegistrationStartRequest;
+import it.smartcommunitylab.aac.webauthn.provider.WebAuthnIdentityProviderConfig;
+import it.smartcommunitylab.aac.webauthn.provider.WebAuthnIdentityService;
+import it.smartcommunitylab.aac.webauthn.provider.WebAuthnSubjectResolver;
+import it.smartcommunitylab.aac.webauthn.service.WebAuthnRpService;
+import it.smartcommunitylab.aac.webauthn.service.WebAuthnRpServiceRegistrationRepository;
+
+/**
+ * Manages the endpoints connected to the registration ceremony of WebAuthn.
+ * 
+ * The registration ceremony is used to register new WebAuthn credentials on
+ * this server.
+ */
+@Controller
+@RequestMapping
+public class WebAuthnRegistrationController {
+
+    private final ProviderRepository<WebAuthnIdentityProviderConfig> registrationRepository;
+
+    public WebAuthnRegistrationController(
+            ProviderRepository<WebAuthnIdentityProviderConfig> registrationRepository) {
+        this.registrationRepository = registrationRepository;
+    }
+
+    private ObjectMapper mapper = new ObjectMapper();
```

Comment:
drop the mapper, as already noted the conversion to string should happen in WebAuthnAttestationResponse

Reply:
Done in 9a53d7b7d145fe5eb8309cca2f9fe171746b8a90. Mapper fully removed at 9929d9e09c08dec9313256ac20b119c0f59fbadb